src/catalog/catalog_set.cpp

```diff
@@ -211,7 +211,7 @@ CatalogEntrySet CatalogSet::getEntries(const Transaction* transaction) {
 
 void CatalogSet::iterateEntriesOfType(const Transaction* transaction, CatalogEntryType type,
     const std::function<void(CatalogEntry*)>& func) {
-    std::lock_guard lck{mtx};
+    std::shared_lock lck{mtx};
```

Comment:
I don't think shared_lock is safe here, as the `func` can modify entires.
One option is to change func signature to take in `const CatalogEntry&` instead of `CatalogEntry*`, or we should fall back to `unique_lock` here. Can you see if you can apply the first solution?

Reply:
👍 ,Done